### ISSUE TITLE
CI 2.2.0 - Unexpected Session Timeout/Termination

### ISSUE THREAD

**Issue by bartmika** [bcit-ci/CodeIgniter#3470]
Hi!

We are having a problem with sessions; more specifically, they unexpected timeout.

Our config is setup as such.
![screen shot 2015-01-12 at 3 23 01 pm](https://cloud.githubusercontent.com/assets/5767602/5710313/0ce6dcd6-9a6f-11e4-9e02-6a524808d66b.png)

Our database had the ci_session table created as:
![screen shot 2015-01-12 at 3 23 28 pm](https://cloud.githubusercontent.com/assets/5767602/5710324/2b9c0c46-9a6f-11e4-9d2f-7108427ea7f5.png)

And our encryption key is some random number.

Also, we use a lot of AJAX calls; somewhere, between 5-20 calls.

May you investigate? Thank you!

**Comment by bartmika** [comment]
Hmmm, this looks strinkinly similar to issue #154. Will apply solution from there and close this ticket if that solution works.

**Comment by bartmika** [comment]
Can't close ticket, @dordal solution & @sergio-ascanio solution in #154 did not fix the problem.

**Comment by andrefrcastro** [comment]
Did you try to match useragent instead of matching ip ?

```sess_match_ip FALSE	TRUE/FALSE (boolean)	```
> Whether to match the user's IP address when reading the session data. Note that some ISPs dynamically changes the IP, so if you want a non-expiring session you will likely set this to FALSE.

**Comment by bartmika** [comment]
I went into my projects framework url: /system/libraries/Session.php and made the following modification
```php
function sess_update()
	{
        if ( ($this->CI->input->is_ajax_request()) ) { // Added this there lines
            return;
        }

      // THE REST OF THE FUNCTION IS LEFT INTACT.
}
```

And this change appears to fix the problem. I will continue testing and either close this ticket or specific if this problem persists.

@andrefrcastro Thanks for suggestion, I've set it to TRUE and will see if it helps.

**Comment by andrefrcastro** [comment]
I deleted my comment, because i noticed you have changed your config settings in the meanwhile and it wouldn't make any sense.

**Comment by bartmika** [comment]
Nope, problem still persists. This problem appears less frequent, but it still occurs.

**Comment by andrefrcastro** [comment]
Is it possible to share more details of your setup, environment, etc ?

**Comment by narfbg** [comment]
> Nope, problem still persists. This problem appears less frequent, but it still occurs.

Once every 5 minutes?
Set 'sess_time_to_update' to 0, until CI 3.0 is released, which will fix the problem.

**Comment by bartmika** [comment]
Sorry @andrefrcastro, it's complicated to post.

Thanks @narfbg, we will be setting to zero and will close ticket if confirmed working. Also, we plan to port entirely to CI 3.0 once the issue: http://forum.codeigniter.com/thread-744.html is fixed.

**Comment by narfbg** [comment]
Just to clarify, because you seem to misunderstand how our handling of github issues works ... This is not a place for "support tickets". It's still open only because it's a legitimate bug that has been reported over and over again and it's already resolved in the development branch, but I don't want to close another 10 of them.

**Comment by bartmika** [comment]
Understood, thanks for your help @narfbg. 

After all day usage, it appears setting 'sess_time_to_update' to 0 resolves the problem.

**Comment by narfbg** [comment]
Resolved by #3073.
